src/geometryUtils.cpp

```diff
@@ -11,46 +10,46 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+#include <stdlib.h>
 
 #include "src/geometryUtils.h"
 
 namespace wsiToDicomConverter {
 
-void dimensionDownsampling(int64_t frameWidht, int64_t frameHeight,
-                           int64_t levelWidht, int64_t levelHeight, bool retile,
-                           int level, double downsampleOfLevel,
-                           int64_t *frameWidhtDownsampled,
-                           int64_t *frameHeightDownsampled,
-                           int64_t *levelWidhtDownsampled,
-                           int64_t *levelHeightDownsampled) {
-  *frameWidhtDownsampled = frameWidht;
+void dimensionDownsampling(
+    int64_t frameWidth, int64_t frameHeight, int64_t levelWidth,
+    int64_t levelHeight, bool retile, int level, double downsampleOfLevel,
+    int64_t *frameWidthDownsampled, int64_t *frameHeightDownsampled,
+    int64_t *levelWidthDownsampled, int64_t *levelHeightDownsampled,
+    int64_t *level_frameWidth, int64_t *level_frameHeight) {
+  *frameWidthDownsampled = frameWidth;
   *frameHeightDownsampled = frameHeight;
-  *levelWidhtDownsampled = levelWidht;
+  *levelWidthDownsampled = levelWidth;
   *levelHeightDownsampled = levelHeight;
+  *level_frameWidth = frameWidth;
+  *level_frameHeight = frameHeight;
   if (retile && level > 0) {
-    *frameWidhtDownsampled *= downsampleOfLevel;
+    *frameWidthDownsampled *= downsampleOfLevel;
     *frameHeightDownsampled *= downsampleOfLevel;
-    *levelWidhtDownsampled /= downsampleOfLevel;
+    *levelWidthDownsampled /= downsampleOfLevel;
     *levelHeightDownsampled /= downsampleOfLevel;
   }
-  if (levelWidht <= *frameWidhtDownsampled &&
-      levelHeight <= *frameHeightDownsampled) {
-    if (levelWidht >= levelHeight) {
-      adjustFrameToLevel(frameWidhtDownsampled, frameHeightDownsampled,
-                         levelWidht);
-    } else {
-      adjustFrameToLevel(frameHeightDownsampled, frameWidhtDownsampled,
-                         levelHeight);
-    }
+  if (levelWidth < *frameWidthDownsampled) {
+    /*
+        level_frameWidth could optimally be set to *levelWidthDownsampled
+        However, a bug in JPEG2000 Codec segfaults if the size of the allocated
+       frame being compressed is too small.  JPEG Codec is fine.  The max is a
+       bandaid fix for the isseue. JPEG2000 codec should be fixed and the max
+       can be removed here and in levelHeight < To reproduce bug: remove max,
+       run JPEG2000 test case in end-to-end tests.
+    */
+    *frameWidthDownsampled = levelWidth;
+    *level_frameWidth = std::max<int64_t>(*levelWidthDownsampled, 50);
+  }
+  if (levelHeight < *frameHeightDownsampled) {
+    *frameHeightDownsampled = levelHeight;
+    *level_frameHeight = std::max<int64_t>(*levelHeightDownsampled, 50);
```

Comment:
Not very familiar with this code, could you add a bit more explanation on these changes and what the bug you're fixing is? Why do frame width and height need to be updated here? From what I understand these two parameters are used to set the final dicom frame width and height by rescaling the boost view in frame.cpp